src/index.js

```diff
@@ -301,10 +311,10 @@ class MuiTable extends Component {
             }
             style={{ width: 'inherit' }} // fix text overflowing
             direction={orderDirection}
-            onClick={() =>
+            onClick={(event) =>
               column.onHeaderClick
-                ? column.onHeaderClick()
-                : onHeaderClick(column)
+                ? column.onHeaderClick(event)
+                : onHeaderClick(column, event)
```

Comment:
This should pass `event` as the first argument, and `column` within an object.

```suggestion
                : onHeaderClick(event, { column })
```